src/supervisor.ts

```diff
@@ -40,25 +40,24 @@ export class Supervisor {
 		this.config = new Config({ db: this.db });
 		this.eventTracker = new EventTracker();
 		this.logger = new Logger({ db: this.db, eventTracker: this.eventTracker });
-		this.deviceState = new DeviceState({
+		this.apiBinder = new APIBinder({
 			config: this.config,
 			db: this.db,
+			deviceState: this.deviceState,
 			eventTracker: this.eventTracker,
 			logger: this.logger,
 		});
-		this.apiBinder = new APIBinder({
+		this.deviceState = new DeviceState({
 			config: this.config,
 			db: this.db,
-			deviceState: this.deviceState,
 			eventTracker: this.eventTracker,
 			logger: this.logger,
+			apiBinder: this.apiBinder,
 		});
 
 		// FIXME: rearchitect proxyvisor to avoid this circular dependency
 		// by storing current state and having the APIBinder query and report it / provision devices
 		this.deviceState.applications.proxyvisor.bindToAPI(this.apiBinder);
-		// We could also do without the below dependency, but it's part of a much larger refactor
-		this.deviceState.applications.apiBinder = this.apiBinder;
```

Comment:
I needed to refactor this so that ApplicationManager knew about its `apiBinder` property otherwise it wouldn't exist in the inferred types which caused a bunch of headaches, so that's where all the `apiBinder` related changes come from (required by improved types)